app.rb

```diff
@@ -90,42 +92,62 @@ def create_book
   end
 
   def create_rental
-    if @book_list.empty?
-      puts 'No book record found'
-    elsif @people_list.empty?
-      puts 'No person record found'
-    else
-      puts 'Select a book from the following list by number '
-      @book_list.each_with_index do |book, index|
-        puts "#{index}) Title: #{book.title}, Author: #{book.author}"
-      end
+    return puts 'No book records found' if @book_list.empty?
+    return puts 'No person records found' if @people_list.empty?
 
-      book_index = gets.chomp.to_i
+    display_books
+    book_index = gets.chomp.to_i
 
-      puts 'Select a person from the following list by number '
-      @people_list.each_with_index do |person, index|
-        puts "#{index}) [#{person.class}] Name : #{person.name}, ID: #{person.id}, Age: #{person.age}"
-      end
+    display_people
+    person_index = gets.chomp.to_i
 
-      person_index = gets.chomp.to_i
+    print 'Date: '
+    date = gets.chomp
 
-      print 'Date: '
-      date = gets.chomp
+    rental = Rental.new(date, @book_list[book_index], @people_list[person_index])
+    @rental_list << rental
 
-      @rental_list << Rental.new(date, @book_list[book_index], @people_list[person_index])
-      puts 'Rental created successfully '
-    end
+    puts 'Rental created successfully'
   end
 
   def list_rentals
-    print 'ID of person : '
+    if @rental_list.empty?
+      puts 'No rental records found'
+      return
+    end
+
+    print 'ID of person: '
     id = gets.chomp.to_i
 
     rentals = @rental_list.filter { |rental| rental.person.id == id }
 
-    puts 'Rentals :'
-    rentals.each do |rental|
-      puts "Date : #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+    if rentals.empty?
+      puts 'No rentals found for the specified person ID'
+    else
+      puts 'Rentals:'
+      rentals.each do |rental|
+        puts "Date: #{rental.date}, Book '#{rental.book.title}' by #{rental.book.author}"
+      end
+    end
+  end
```

Comment:
- [ ] this is what the previous CR means with:
 _"Other logics such as requests for input (get.chomps) or input validations\print statements should be placed in the main.rb, separate classes or methods."_
In this case I suggest you create a separate class (named Input for example) with the methods person_input, book_input, rental_input and however many you think you need and put all the `print` and `gets` statements in these methods. after that in the App class you just call these methods instead of printing, getting and processing this data directly in this class.